diff --git a/rofi-power-menu b/rofi-power-menu
index 13e17e3..11ce0a2 100755
--- a/rofi-power-menu
+++ b/rofi-power-menu
@@ -37,8 +37,13 @@ icons[reboot]="\Uf0709"
 icons[shutdown]="\Uf0425"
 icons[cancel]="\Uf0156"
 
+function hyprlock_action {
+    hyprlock > /dev/null 2>&1 &
+    disown
+}
+
 declare -A actions
-actions[lockscreen]="loginctl lock-session ${XDG_SESSION_ID-}"
+actions[lockscreen]="hyprlock_action"
 #actions[switchuser]="???"
 actions[logout]="loginctl terminate-session ${XDG_SESSION_ID-}"
 actions[suspend]="systemctl suspend"
